### ISSUE TITLE
missed ability to handle invalid GET requests

### ISSUE THREAD

**Issue by Fivell** [swisnl/json-api-client#39]
## Detailed description

Hello guys take a look at following implementation
https://github.com/swisnl/json-api-client/blob/master/src/Repository.php#L44
and
https://github.com/swisnl/json-api-client/blob/master/src/Repository.php#L84

For some reason Repository class raises an exception that doesn't allow to handle error document errors

By json.api spec get request can return error documents

for instance

https://jsonapi.org/format/#fetching-includes

```
If an endpoint does not support the include parameter, it MUST respond with 400 Bad Request to any requests that include it.
```

same for invalid filters, sorting etc...

I think repository class should be changed to raise Exception that has public property of Error instance that has  description describes that include for such relation is not supported or filter is not valid or something else that server is responding.

To summarize: 
Would be good to have ability to see user friendly error message for 400 responses that actually are valid jsonapi document errors, implemented by https://github.com/swisnl/json-api-client/blob/master/src/Errors/Error.php

**Comment by Fivell** [comment]
another option is to remove checks for instance of Document at all
https://github.com/swisnl/json-api-client/blob/master/src/Repository.php#L44
and
https://github.com/swisnl/json-api-client/blob/master/src/Repository.php#L84

 and allow such code to work

```
$blogsDocument = BlogRepository.all(['filter' => '-createdAt'];
if $blogsDocument->hasErrors()
{
   //do something with errors
} 
else
{
  $blogs = $blogsDocument->getData();
}
```

For now Get responses are handled in its own way and raises custom exceptions, for other Requests types developer has  access to errors

**Comment by JaZo** [comment]
We've been thinking about removing the exceptions from the Repository for a while as it is up to the consumer to determine the error behaviour and the current implementation does not allow custom checks. We hadn't made a decision yet, but I just created a PR to push this through.
